bookshelf/tests/api_v3/test_cloud.py

```diff
@@ -0,0 +1,339 @@
+    """
+    _ENV_VAR = 'ACCEPTANCE_YAML'
+    filename = os.environ.get(_ENV_VAR)
+    if not filename:
+        print (
+            'Must set {} to an acceptance.yaml file ('
+            'http://doc-dev.clusterhq.com/gettinginvolved/appendix.html#acceptance-testing-configuration'  #noqa
+            ') plus additional keys in order to run this test.'.format(
+                _ENV_VAR))
+        raise unittest.SkipTest()
+    with open(filename) as f:
+        config = yaml.safe_load(f)
+    return config
+
+
+def _get_yaml_config(substructure, config=None):
+    """
+    Extract the keys from the config in substructure, which may be a nested
+    dictionary.
+
+    Raises a ``unittest.SkipTest`` if the substructure is not found in the
+    configuration.
+
+    This can be used to load credentials all at once for testing purposes.
+    """
+    if config is None:
+        config = _load_config_from_yaml()
+    try:
+        return _extract_substructure(config, substructure)
+    except MissingConfigError as e:
+        print (
+            'Skipping test: could not get configuration: {}\n\n'
+            'In order to run this test, add ensure file at $ACCEPTANCE_YAML '
+            'has structure like:\n\n{}'.format(
+                e.message,
+                yaml.dump(substructure, default_flow_style=False)))
+        raise unittest.SkipTest()
+
+
+class RackspaceTests(unittest.TestCase, CloudInstanceTestMixin):
+    """
+    Tests for rackspace.
+    """
+
+    def setUp(self):
+        super(RackspaceTests, self).setUp()
+
+        credentials = _get_yaml_config(
+            {
+                'rackspace': {
+                    'keyname': '<Rackspace keypair name>',
+                    'username': '<Rackspace username>',
+                    'key': '<Rackspace API key>',
+                },
+                'ssh_keys': {
+                    'rackspace': {
+                        'private_key_file': '<path-to-private-keypair-file>',
+                        'public_key_file': '<path-to-corresponding-public-key>'
+                    }
+                }
+            }
+        )
+
+        ssh_keys = credentials["ssh_keys"]["rackspace"]
+        self.config = RackspaceConfiguration(
+            username='root',
+            instance_type='1GB Standard Instance',
+            key_pair=credentials["rackspace"]["keyname"],
+            public_key_filename=ssh_keys["public_key_file"],
+            private_key_filename=ssh_keys["private_key_file"],
+            access_key_id=credentials["rackspace"]["username"],
+            secret_access_key=credentials["rackspace"]["key"],
+            ami='CentOS 7 (PVHVM)',
+            description='rackspace-test-instance-description',
+            image_basename='rackspace-test-image',
+            instance_name='rackspace-test-instance'
+        ).serialize()
+        self.distribution = Distribution.CENTOS7
+        self.region = 'HKG'
+        self.instance_factory = RackspaceInstance
```

Comment:
HKG is a good default as not many people I know have been there, but it would be cool if I could change these from the config.yam file